src/models/transactions/transaction.ts

```diff
@@ -44,3 +47,98 @@ export interface TransactionAndMetadata {
     transaction: Transaction
     metadata: Metadata
 }
+
+/**
+ * 
+ * @param {Transaction} tx A Transaction.
+ * @returns {void}
+ * @throws {ValidationError} When the Transaction is malformed.
+ */
+ export function verify(tx: Transaction) {
+    
+  if(!isEqual(decode(encode(tx)),tx))
```

Comment:
Maybe also add a check for `onlyHasFields` (a helper function in `models`) before this equality is checked, as it will also help to provide better error messages